cli/parse_args.ts

```diff
@@ -373,12 +373,11 @@ function setNested(
   value: unknown,
   collect = false,
 ) {
-  keys.slice(0, -1).forEach((key) => {
-    object[key] ??= {};
-    object = object[key] as NestedMapping;
-  });
+  keys = [...keys];
+  const key = keys.pop();
+  if (!key) throw new Error(`'keys' cannot be an empty array.`);
```

Comment:
This check seems always false

Reply:
Ah, yes. Are you suggesting we just use a non-null assertion?

Reply:
ah, wait, it might be fine. Let me check

Reply:
> Ah, yes. Are you suggesting we just use a non-null assertion?

Sounds good. That was what the previous code was doing at `const key = keys.at(-1)!`